pkg/storage/unified/resource/quotas.go

```diff
@@ -0,0 +1,175 @@
+	// shouldn't be empty since we use file path existence to determine if we should enable the service
+	if opts.FilePath == "" {
+		return nil, fmt.Errorf("quota overrides file path is required")
+	}
+	if opts.ReloadPeriod == 0 {
+		opts.ReloadPeriod = time.Second * 30
+	}
+
+	// Check if file exists
+	if _, err := os.Stat(opts.FilePath); err != nil {
+		if os.IsNotExist(err) {
+			return nil, fmt.Errorf("quota overrides file does not exist: %s", opts.FilePath)
+		}
+		return nil, fmt.Errorf("failed to stat quota overrides file: %w", err)
+	}
+
+	config := runtimeconfig.Config{
+		ReloadPeriod: opts.ReloadPeriod,
+		LoadPath:     []string{opts.FilePath},
+		Loader: func(r io.Reader) (interface{}, error) {
+			var tenants map[string]TenantQuotas
+			decoder := yaml.NewDecoder(r)
+			if err := decoder.Decode(&tenants); err != nil {
+				return nil, err
+			}
+			return &QuotaOverrides{Tenants: tenants}, nil
+		},
+	}
+
+	manager, err := runtimeconfig.New(config, "custom-quotas", reg, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	return &QuotaService{
+		manager: manager,
+		logger:  logger,
+	}, nil
+}
+
+// one the runtimeconfig manager is in a running state, it will periodically reload the configuration file into the manager if there are changes
+func (q *QuotaService) init(ctx context.Context) error {
+	err := q.manager.StartAsync(ctx)
+	if err != nil {
+		return err
+	}
+	err = q.manager.AwaitRunning(ctx)
+	if err != nil {
+		return err
+	}
+
+	// set initial config once running
+	q.SetConfig(q.manager.GetConfig().(*QuotaOverrides))
+
+	// update config on changes
+	updateChannel := q.manager.CreateListenerChannel(1)
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case update := <-updateChannel:
+				q.logger.Info("Received quota config update", "update", update)
+				newConfig, ok := update.(*QuotaOverrides)
+				if !ok {
+					q.logger.Error("Failed to convert quota config update to QuotaOverrides", "update", update)
+					continue
+				}
+				q.SetConfig(newConfig)
+			}
+		}
+	}()
+
+	return nil
+}
+
+func (q *QuotaService) stop() {
+	q.manager.StopAsync()
+}
+
+func (q *QuotaService) GetConfig() *QuotaOverrides {
+	q.overridesMutex.Lock()
+	defer q.overridesMutex.Unlock()
+	return q.quotaOverrides
+}
+
+func (q *QuotaService) SetConfig(overrides *QuotaOverrides) {
+	q.overridesMutex.Lock()
+	q.quotaOverrides = overrides
+	q.overridesMutex.Unlock()
+}
+
+func (q *QuotaService) GetQuota(nsr NamespacedResource) (ResourceQuota, error) {
+	if nsr.Namespace == "" || nsr.Resource == "" || nsr.Group == "" {
+		return ResourceQuota{}, fmt.Errorf("invalid namespaced resource: %+v", nsr)
+	}
+
+	overrides := q.GetConfig()
+	// should never be nil - but just in case
+	if overrides == nil {
+		return ResourceQuota{Limit: DEFAULT_RESOURCE_LIMIT}, nil
+	}
+
+	tenantId := strings.TrimPrefix(nsr.Namespace, "stacks-")
```

Comment:
The docs at the top of `NewQuotaService` mention a YAML file with the `stacks-` prefix, but we are removing it here, so maybe that comment needs updating?

Reply:
good catch! fixed 👍